plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/cleaner/DockerAbandonedResourcesCleaner.java

```diff
@@ -15,52 +15,85 @@
 
 /**
- * Job for periodically clean up inactive docker containers and log list active containers.
+ * Job for periodically clean up abandoned docker containers and networks created by CHE.
+ * Also, logs active containers list.
  *
  * @author Alexander Andrienko
+ * @author Mykola Morhun
  */
 @Singleton
-public class DockerContainerCleaner implements Runnable {
+public class DockerAbandonedResourcesCleaner implements Runnable {
 
-    private static final Logger LOG = LoggerFactory.getLogger(DockerContainerCleaner.class);
+    private static final Logger LOG = LoggerFactory.getLogger(DockerAbandonedResourcesCleaner.class);
+
+    private static final Filters           NETWORK_FILTERS          = new Filters().withFilter("type", "custom");
+    private static final GetNetworksParams GET_NETWORKS_PARAMS      = GetNetworksParams.create().withFilters(NETWORK_FILTERS);
+    private static final String            WORKSPACE_ID_REGEX_GROUP = "workspaceId";
+    private static final String            CHE_NETWORK_REGEX        = "^(?<" + WORKSPACE_ID_REGEX_GROUP + ">workspace[a-z\\d]{16})_[a-z\\d]{16}$";
+    private static final Pattern           CHE_NETWORK_PATTERN      = Pattern.compile(CHE_NETWORK_REGEX);
 
     // TODO replace with WorkspaceManager
     private final CheEnvironmentEngine         environmentEngine;
     private final DockerConnector              dockerConnector;
     private final DockerContainerNameGenerator nameGenerator;
+    private final WorkspaceRuntimes            runtimes;
+    private final Set<String>                  additionalNetworks;
 
     @Inject
-    public DockerContainerCleaner(CheEnvironmentEngine environmentEngine,
-                                  DockerConnector dockerConnector,
-                                  DockerContainerNameGenerator nameGenerator) {
+    public DockerAbandonedResourcesCleaner(CheEnvironmentEngine environmentEngine,
+                                           DockerConnector dockerConnector,
+                                           DockerContainerNameGenerator nameGenerator,
+                                           WorkspaceRuntimes workspaceRuntimes,
+                                           @Named("machine.docker.networks") Set<Set<String>> additionalNetworks) {
         this.environmentEngine = environmentEngine;
         this.dockerConnector = dockerConnector;
         this.nameGenerator = nameGenerator;
+        this.runtimes = workspaceRuntimes;
+        this.additionalNetworks = additionalNetworks.stream()
+                                                    .flatMap(Set::stream)
+                                                    .collect(toSet());
     }
 
-    @ScheduleRate(periodParameterName = "che.docker.unused_containers_cleanup_min",
-                  initialDelayParameterName = "che.docker.unused_containers_cleanup_min",
+    @ScheduleRate(periodParameterName = "che.docker.cleanup_period_min",
+                  initialDelay = 0L,
```

Comment:
Use-case: a user had many workspaces running and then his PC hanged up, he reset it and run CHE again. It can lead to some troubles if we leave all that things as is. `initialDelay = 0` runs cleaner on server startup.